js/index.js

```diff
@@ -1,100 +1,360 @@
 
 webix.ready(function(){
 
-  var grid_data = [
-    {id:1, title:"The Shawshank Redemption", year:1994, votes:678790, rating:9.2, rank:1},
-    {id:2, title:"The Godfather", year:1972, votes:511495, rating:9.2, rank:2},
-    {id:3, title:"The Godfather: Part II", year:1974, votes:319352, rating:9.0, rank:3},
-    {id:4, title:"The Good, the Bad and the Ugly", year:1966, votes:213030, rating:8.9, rank:4},
-    {id:5, title:"My Fair Lady", year:1964, votes:533848, rating:8.9, rank:5},
-    {id:6, title:"12 Angry Men", year:1957, votes:164558, rating:8.9, rank:6}
-  ];
-
-  webix.ui({
+  var dashboard_data_link = "./data/data.js";
+  var users_data_link = "./data/users.js";
+  var products_data_link = "./data/products.js";
 
+  var topbar = {
+    view:"toolbar",
+    css:"webix_dark",
+    elements:[
+      {view: "label", label:"My App"},
+      {
+         view:"button",
+         id:"btn_profile",
+         type:"icon",
+         icon:"wxi-user",
+         label:"Profile",
+         css:"webix_transparent",
+         width:100,
+         popup:"profile_popup"
+      }
+    ]
+  };
+  var sidebar = {
+    type:"clean",
+    width:300,
+    minWidth:250,
+    maxWidth:350,
     rows:[
+      {
+        view:"list",
+        id:"side_menu_list",
+        scroll:false,
+        css:"gray_background",
+        select:true,
+        on:{
+          onAfterSelect:function(id){
+            $$(id).show();
+          }
+        },
+        data:[ "Dashboard", "Users", "Products", "Admin"]
+      },
+      {
+        view:"label",
+        label: "<span class='green_text webix_icon mdi mdi-check'></span><span class='green_text'>Connected</span>",
+        align:"center",
+        height:40,
+        css:"gray_background",
+      }
+    ]
+  };
+  var data = {
+    view:"datatable",
+    id:"films_datatable",
+    scrollY:true,
+    minWidth:400,
+    columns:[
+      { id:"rank", header:"", width:50, css:"rank", sort:"int"},
+      { id:"title", header:["Film title",{ content:"textFilter"}], fillspace:true, sort:"text"},
+      { id:"year", header:["Year",{ content:"textFilter"}], width:80, sort:"int"},
+      { id:"rating", header:["Rating",{ content:"selectFilter"}], width:80, sort:"int"},//sort of Rating doesn't work correctly!!!
+      { id:"votes", header:["Votes",{ content:"textFilter"}], width:80, sort:"int" //sort of Votes doesn't work correctly!!!
```

Comment:
the problem is within the incoming data, which needs to be resolved in UI: 

- define `scheme` and convert string data into valid numbers before loading (it is from next lesson, though)
- format numbers in a Datatable column using template or format setting

Reply:
I have used several ways solving this problem such as:
"template:(obj)=>{return `${Number(obj.votes.replace(",", "."))}`}" or
"template:function(obj, common){return webix.i18n.numberFormat(obj.votes);}"
- edited data and data format in cell. Visually, I got good result, but sorting was bad.

Reply:
The trick does [data scheme](https://docs.webix.com/desktop__data_scheme.html#initkey)
It parses strings to numbers during loading

then template just defines how this numbers will look like, while for sorting numbers will be used 

https://snippet.webix.com/yxucm596

Reply:
The sorting problem was solved.
I have used next:
    view:"datatable",
    scheme:{
        $init:function(obj){
            obj.votes = Number(obj.votes.replace(",", "."));
            obj.rating = Number(obj.rating.replace(",", "."));
            obj.year = Number(obj.year);
        }
    }